Sources/ProjectDescription/MultiplatformTarget.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+
+extension Multiplatform {
```

Comment:
The approach you've taken here is much cleaner though so I see the value in that, main concern is maintaining two duplicate `Target`s 

Would it be possible to make this the main `Target` structure but map the the legacy properties to the appropriate underlying new properties?

e.g.

```swift
init(
// ...
  platform: Platform,
  deploymentTarget: DeploymentTarget,
// ...
) {
  self.init(
   // ...
     destinations: [Destination.from(platform)],
     deploymentTargets: DeploymentTargets.from(deploymentTarget)
   // ...
  )
}
```

Reply:
yes we can and should.  this approach was taken to continue to support source compatibility for current Tuist users.  given that `platform` is a public property of `Target`, I didn't feel safe changing how things like that are exposed at this point in time.

 If we're open to breaking changes, we can very much adjust the main `Target` type for our new needs as well as provide a convenience initializer that makes migration simpler for folks that dont need multiplatform support.

Reply:
Makes sense to maintain source compatibility.

My thinking is we may be able to map / derive the old properties to prevent breaking changes, however if that proves to be too much hassle the approach you have taken is much cleaner.

Reply:
I'm on the fence here. While supporting source compatibility is important, the cost of maintaining two similar structures in the codebase creates room for inconsistencies. I'm not a big fan of breaking changes, but multi-platform support is an important change for Tuist and might warrant going with a major release, in which we can batch other deprecated APIs. If I'm not wrong here, the changes developers would have to make are not that significant, so even if it's a breaking change, which is always undesirable, the process "should" be straightforward.

Reply:
My main thought for this was that it would be "short" lived.  We could publish the source compatible version for a while, allow users to migrate at their own pace and incrementally if they have many targets to migrate.   After enough time we can make the source breaking change to remove the old deprecated target implementation and promote the multiplatform one to the only one.